In [1]:
import pandas as pd

import warnings
warnings.filterwarnings('ignore')

In [2]:
# Importing the metadata that does not have continents and subregions only countries
metadata_country =pd.read_csv(r"C:\Users\chine\Downloads\Cleaned Data\Cleaned MetaData Country.csv" ,encoding ="latin-1")
metadata_country

country_name Country Code                      Region  \
0           Aruba          ABW   Latin America & Caribbean   
1     Afghanistan          AFG                  South Asia   
2          Angola          AGO          Sub-Saharan Africa   
3         Albania          ALB       Europe & Central Asia   
4         Andorra          AND       Europe & Central Asia   
..            ...          ...                         ...   
212        Kosovo          XKX       Europe & Central Asia   
213   Yemen, Rep.          YEM  Middle East & North Africa   
214  South Africa          ZAF          Sub-Saharan Africa   
215        Zambia          ZMB          Sub-Saharan Africa   
216      Zimbabwe          ZWE          Sub-Saharan Africa   

             IncomeGroup                                       SpecialNotes  
0            High income                                                NaN  
1             Low income  The reporting period for national accounts dat...  
2    Lower middle income  The World Bank systematically assesses the app...  
3    Upper middle income                                                NaN  
4            High income                                                NaN  
..                   ...                                                ...  
212  Upper middle income                                                NaN  
213           Low income  The World Bank systematically assesses the app...  
214  Upper middle income  Fiscal year end: March 31; reporting period fo...  
215  Lower middle income  National accounts data were rebased to reflect...  
216  Lower middle income  National Accounts data are reported in Zimbabw...  

[217 rows x 5 columns]

In [3]:
# checking the total number of the countries in metadata
metadata_country['country_name'].nunique()

217

In [4]:
remittance= pd.read_csv(r"C:\Users\chine\Downloads\Cleaned Data\Remittance 1970 - 2022.csv")
remittance

country_name country_code  Year  remittance_received  \
0                            Aruba          ABW  1970         0.000000e+00   
1      Africa Eastern and Southern          AFE  1970         2.265999e+07   
2                      Afghanistan          AFG  1970         0.000000e+00   
3       Africa Western and Central          AFW  1970         0.000000e+00   
4                           Angola          AGO  1970         0.000000e+00   
...                            ...          ...   ...                  ...   
14093                       Kosovo          XKX  2022         1.612854e+09   
14094                  Yemen, Rep.          YEM  2022         0.000000e+00   
14095                 South Africa          ZAF  2022         8.728556e+08   
14096                       Zambia          ZMB  2022         2.434860e+08   
14097                     Zimbabwe          ZWE  2022         2.047406e+09   

       remittance_paid        gdp  unemployment_rate  
0         0.000000e+00   0.000000              0.000  
1         2.365999e+08   0.000000              0.000  
2         0.000000e+00   0.000000              0.000  
3         0.000000e+00   0.000000              0.000  
4         0.000000e+00   0.000000              0.000  
...                ...        ...                ...  
14093     6.071223e+07  17.104969              0.000  
14094     0.000000e+00   0.000000             13.590  
14095     1.011890e+09   0.215058             29.806  
14096     1.312541e+08   0.817494              6.131  
14097     0.000000e+00   9.901347              7.945  

[14098 rows x 7 columns]

In [5]:
# checking the total number of the remittance data
remittance['country_name'].nunique()

266

There are 217 countries in the metadata  But we have 266  in the remittance data set
We need to find the countries that are not supposed to be in the remittance data

In [6]:
# Getting the names of the countries in the remittance datafarme and storing them in a list
remittance_country_list = remittance['country_name'].unique()
remittance_country_list

array(['Aruba', 'Africa Eastern and Southern', 'Afghanistan',
       'Africa Western and Central', 'Angola', 'Albania', 'Andorra',
       'Arab World', 'United Arab Emirates', 'Argentina', 'Armenia',
       'American Samoa', 'Antigua and Barbuda', 'Australia', 'Austria',
       'Azerbaijan', 'Burundi', 'Belgium', 'Benin', 'Burkina Faso',
       'Bangladesh', 'Bulgaria', 'Bahrain', 'Bahamas, The',
       'Bosnia and Herzegovina', 'Belarus', 'Belize', 'Bermuda',
       'Bolivia', 'Brazil', 'Barbados', 'Brunei Darussalam', 'Bhutan',
       'Botswana', 'Central African Republic', 'Canada',
       'Central Europe and the Baltics', 'Switzerland', 'Channel Islands',
       'Chile', 'China', "Cote d'Ivoire", 'Cameroon', 'Congo, Dem. Rep.',
       'Congo, Rep.', 'Colombia', 'Comoros', 'Cabo Verde', 'Costa Rica',
       'Caribbean small states', 'Cuba', 'Curacao', 'Cayman Islands',
       'Cyprus', 'Czechia', 'Germany', 'Djibouti', 'Dominica', 'Denmark',
       'Dominican Republic', 'Algeria',
 

In [7]:
len(remittance_country_list)

266

In [8]:
# Lets get the unique names of countries in metadata  and Create a list to store the  countries 
metadata_country_list = metadata_country['country_name'].unique() 
metadata_country_list

array(['Aruba', 'Afghanistan', 'Angola', 'Albania', 'Andorra',
       'United Arab Emirates', 'Argentina', 'Armenia', 'American Samoa',
       'Antigua and Barbuda', 'Australia', 'Austria', 'Azerbaijan',
       'Burundi', 'Belgium', 'Benin', 'Burkina Faso', 'Bangladesh',
       'Bulgaria', 'Bahrain', 'Bahamas, The', 'Bosnia and Herzegovina',
       'Belarus', 'Belize', 'Bermuda', 'Bolivia', 'Brazil', 'Barbados',
       'Brunei Darussalam', 'Bhutan', 'Botswana',
       'Central African Republic', 'Canada', 'Switzerland',
       'Channel Islands', 'Chile', 'China', "Cote d'Ivoire", 'Cameroon',
       'Congo, Dem. Rep.', 'Congo, Rep.', 'Colombia', 'Comoros',
       'Cabo Verde', 'Costa Rica', 'Cuba', 'Curacao', 'Cayman Islands',
       'Cyprus', 'Czechia', 'Germany', 'Djibouti', 'Dominica', 'Denmark',
       'Dominican Republic', 'Algeria', 'Ecuador', 'Egypt, Arab Rep.',
       'Eritrea', 'Spain', 'Estonia', 'Ethiopia', 'Finland', 'Fiji',
       'France', 'Faroe Islands', 'Micronesia, Fed

In [9]:
len(metadata_country_list)

217

In [10]:
# Lets confirm if the number of coutries we have in our remittance dataset is equal to the number of countries
# we have metadata set
remittance_country_list == metadata_country_list

False

In [11]:
# a dataframe to extract only the countries in metadata country dataset

realcountries = remittance[remittance['country_name'].isin(metadata_country_list)]
realcountries

country_name country_code  Year  remittance_received  remittance_paid  \
0             Aruba          ABW  1970         0.000000e+00     0.000000e+00   
2       Afghanistan          AFG  1970         0.000000e+00     0.000000e+00   
4            Angola          AGO  1970         0.000000e+00     0.000000e+00   
5           Albania          ALB  1970         0.000000e+00     0.000000e+00   
6           Andorra          AND  1970         0.000000e+00     0.000000e+00   
...             ...          ...   ...                  ...              ...   
14093        Kosovo          XKX  2022         1.612854e+09     6.071223e+07   
14094   Yemen, Rep.          YEM  2022         0.000000e+00     0.000000e+00   
14095  South Africa          ZAF  2022         8.728556e+08     1.011890e+09   
14096        Zambia          ZMB  2022         2.434860e+08     1.312541e+08   
14097      Zimbabwe          ZWE  2022         2.047406e+09     0.000000e+00   

             gdp  unemployment_rate  
0       0.000000              0.000  
2       0.000000              0.000  
4       0.000000              0.000  
5       0.000000              0.000  
6       0.000000              0.000  
...          ...                ...  
14093  17.104969              0.000  
14094   0.000000             13.590  
14095   0.215058             29.806  
14096   0.817494              6.131  
14097   9.901347              7.945  

[11501 rows x 7 columns]

In [12]:
# checking the total number of the remmitance data
realcountries['country_name'].nunique()

217

In [13]:
# Getting the names of the countries that are not metadata country data set
print(set(remittance_country_list) ^ set(metadata_country_list))

{'IDA & IBRD total', 'East Asia & Pacific (excluding high income)', 'Not classified', 'Africa Western and Central', 'South Asia (IDA & IBRD)', 'Europe & Central Asia (excluding high income)', 'Central Europe and the Baltics', 'Pacific island small states', 'Middle East & North Africa (excluding high income)', 'Post-demographic dividend', 'IDA only', 'Africa Eastern and Southern', 'Lower middle income', 'Heavily indebted poor countries (HIPC)', 'Fragile and conflict affected situations', 'European Union', 'North America', 'Middle East & North Africa', 'Low & middle income', 'Low income', 'Arab World', 'High income', 'OECD members', 'Middle East & North Africa (IDA & IBRD countries)', 'Europe & Central Asia', 'IBRD only', 'East Asia & Pacific (IDA & IBRD countries)', 'Latin America & Caribbean', 'Caribbean small states', 'Least developed countries: UN classification', 'Middle income', 'Sub-Saharan Africa (excluding high income)', 'Euro area', 'Early-demographic dividend', 'World', 'South

In [14]:
realcountries.to_csv(r"C:\Users\chine\Downloads\Cleaned Data\Remittance 1970 - 2022 -Chinelo.csv", index = False)   #To save to csv